### PR TITLE
Wallet upgrade for demo

### PR DESCRIPTION
**Pull request by dmitrizagidulin**

### CODE REVIEW

File: packages/server-core/src/createApp.ts [hunk 1]
```diff
@@ -184,6 +185,12 @@ export const createFeathersExpressApp = (
     return res.status(204).send()
   })
 
+  app.post('/api/credentials/request', IssueRequest.post)
```
Comment (HexaField) [inline packages/server-core/src/createApp.ts:188]: is there a reason we are not using feathers for this?

File: packages/engine/src/credentials/CredentialSystem.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+export const CredentialSystem = () => {
+  const didVerifyCredentialActions = createActionQueue(CredentialAction.verificationResult.matches)
+  const nameQuery = defineQuery([NameComponent])
+
+  return () => {
+    for (const entity of nameQuery.enter()) {
+      const name = getComponent(entity, NameComponent).name
+      if (name === 'Door' || name === 'VCKey Trigger Volume') {
```
Comment (HexaField) [inline packages/engine/src/credentials/CredentialSystem.ts:38]: we can't rely on component names here, we should be creating a tag component, and use that to identify an object that is requiring this functionality

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/credentials/CredentialSystem.ts (hunk 2 above)
Comment (dmitrizagidulin) [reply packages/engine/src/credentials/CredentialSystem.ts:38]: @HexaField yeah, this was a temporary hack me and Gheric used for the demo -- we figured we'd discuss with you re how to do it properly

File: packages/engine/src/initializeEngine.ts [hunk 3]
```diff
@@ -54,6 +55,10 @@ export const setupEngineActionSystems = () => {
     type: SystemUpdateType.FIXED_LATE,
     systemFunction: OutgoingActionSystem
   })
+  initSystemSync(world, {
```
Comment (HexaField) [inline packages/engine/src/initializeEngine.ts:58]: system should not be initialized like this, these are core internal engine systems, should be initialized in `initializeCoreSystems` asynchronously instead

File: packages/instanceserver/src/channels.ts [hunk 4]
```diff
@@ -253,6 +254,12 @@ const loadEngine = async (app: Application, sceneId: string) => {
     const sceneResultPromise = app.service('scene').get({ projectName, sceneName, metadataOnly: false }, null!)
 
     await initializeCoreSystems()
+
+    initSystemSync(world, {
```
Comment (HexaField) [inline packages/instanceserver/src/channels.ts:258]: remove this, initialize in initializeCoreSystems

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/initializeEngine.ts (hunk 3 above)
Comment (dmitrizagidulin) [reply packages/engine/src/initializeEngine.ts]: Thanks, moved.

File: packages/instanceserver/src/channels.ts (hunk 4 above)
Comment (dmitrizagidulin) [reply packages/instanceserver/src/channels.ts]: Done.

**Review by speigg — APPROVED** [review]

File: packages/engine/src/credentials/CredentialSystem.ts (hunk 2 above)
Comment (HexaField) [reply packages/engine/src/credentials/CredentialSystem.ts:38]: this still needs to be addressed

File: packages/engine/src/credentials/CredentialSystem.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        addComponent(entity, CallbackComponent, {
+          receiveVC: async () => {
```
Comment (HexaField) [inline packages/engine/src/credentials/CredentialSystem.ts]: lets pull these out into their own functions

File: packages/engine/src/initializeEngine.ts [hunk 6]
```diff
@@ -160,6 +155,11 @@ export const initializeCoreSystems = async (injectedSystems?: SystemModuleType<a
       uuid: 'xre.engine.AssetSystem',
       type: SystemUpdateType.FIXED_LATE,
       systemLoader: () => import('./scene/systems/AssetSystem')
+    },
+    {
+      uuid: 'xre.engine.OutgoingActionSystem',
```
Comment (HexaField) [inline packages/engine/src/initializeEngine.ts:160]: moved the wrong system... OutgoingActionSystem needs to stay where it was

File: packages/engine/src/networking/systems/OutgoingActionSystem.ts [hunk 7]
```diff
@@ -10,7 +10,7 @@ const sendOutgoingActions = (world: World) => {
   }
 }
 
-export default function OutgoingActionSystem(world: World) {
+export default async function OutgoingActionSystem(world: World) {
```
Comment (HexaField) [inline packages/engine/src/networking/systems/OutgoingActionSystem.ts:13]: revert

File: packages/server-core/src/createApp.ts [hunk 8]
```diff
@@ -185,6 +186,12 @@ export const createFeathersExpressApp = (
...
```
Comment (HexaField) [inline packages/server-core/src/createApp.ts:189]: lets use a feathers route instead of a raw REST endpoint

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/initializeEngine.ts (hunk 6 above)
Comment (dmitrizagidulin) [reply packages/engine/src/initializeEngine.ts:160]: Ah, that was a rebasing error, reverting.

File: packages/server-core/src/createApp.ts (hunk 1 above)
Comment (dmitrizagidulin) [reply packages/server-core/src/createApp.ts:189]: Main reason is - I don't know how to do it using feathers. Happy to refactor in the next PR tho.

File: packages/engine/src/credentials/CredentialSystem.ts (hunk 2 above)
Comment (dmitrizagidulin) [reply packages/engine/src/credentials/CredentialSystem.ts:38]: Future PR.

File: packages/engine/src/initializeEngine.ts (hunk 6 above)
Comment (dmitrizagidulin) [reply packages/engine/src/initializeEngine.ts]: Reverted

File: packages/server-core/src/createApp.ts (hunk 8 above)
Comment (dmitrizagidulin) [reply packages/server-core/src/createApp.ts:189]: Don't know how to do that, tho happy to refactor in the next PR.

File: packages/engine/src/credentials/CredentialSystem.ts (hunk 5 above)
Comment (dmitrizagidulin) [reply packages/engine/src/credentials/CredentialSystem.ts]: Extracted

File: packages/projects/default-project/apartment.scene.json [hunk 9]
```diff
@@ -997,12 +997,6 @@
         {
           "name": "visible",
           "props": {}
-        },
```
Comment (HexaField) [inline packages/projects/default-project/apartment.scene.json:1000]: the mount point needs to stay here. if this was due to saving the scene from the editor, then this is a bug that an be addressed elsewhere

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/credentials/CredentialSystem.ts [hunk 10]
```diff
@@ -0,0 +1,73 @@
+import { NotificationService } from '@xrengine/client-core/src/common/services/NotificationService'
```
Comment (HexaField) [inline packages/engine/src/credentials/CredentialSystem.ts:1]: engine cannot be importing client-core, the action should be processed in a separate system in client-core

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/common/src/serverHost.ts [hunk 11]
```diff
@@ -0,0 +1,6 @@
+export const hostDefined = !!globalThis.process.env['VITE_SERVER_HOST']
```
Comment (HexaField) [inline packages/common/src/serverHost.ts:1]: we have these already in client-core/src/util/config.ts lets unify these into a common, as this file has done here

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/projects/default-project/apartment.scene.json (hunk 9 above)
Comment (dmitrizagidulin) [reply packages/projects/default-project/apartment.scene.json:1000]: Yeah, not sure what happened there. Reverting.

**Comment by dmitrizagidulin** [comment]
Ok, the ECS / CredentialSystem stuff is holding up wallet updates. I'm closing this PR and will break this into two separate PRs (put the ecs update into a separate branch).

**Pull request closed without merging**